src/main/java/transforms/CombineDateTimeTransform.java

```diff
@@ -44,10 +44,18 @@ public CombineDateTimeTransform(
         this.datePattern = datePattern;
         this.timeColumn = timeColumn;
         this.timePattern = timePattern;
-        if (timezone.isPresent()){
-            this.timezone = TimeZone.getTimeZone( timezone.get() );
+        if ( timezone.isPresent() ) {
+            String timezoneId = timezone.get();
+
+            this.timezone = Optional.of(TimeZone.getTimeZone( timezoneId ));
+
+            if ( !this.timezone.orElse( Constants.DEFAULT_TIMEZONE ).getID().equals( timezoneId ) ) {
```

Comment:
I think, at this point, `this.timezone` should never be empty, because `TimeZone.getTimeZone( timezoneId )` will fallback to GMT if the id is not valid, so the orElse call is not needed. (also in other places)

Reply:
Also, I would move this whole check on the optional timezone argument into a helper function, because it's used in a lot of places. It could return the optional timezone or throw an error if it's not valid and take the timezone id string plus some custom message for the error message.

Reply:
Wasn't this one of the reasons that the whole time zone issues happened? That getTimeZone was defaulting to GMT?

Reply:
So this is all called within `if ( timezone.isPresent() ) {` so there's no need to do an `orElse( ... ) ` instead of just a `.get()` 

But yeah +1 on the question about why we're using a default timezone at all